activation/activation.go

```diff
@@ -591,23 +611,26 @@ func (b *Builder) PublishActivationTx(ctx context.Context) error {
 func (b *Builder) createAtx(ctx context.Context) (*types.ActivationTx, error) {
 	b.log.With().Info("challenge ready")
 
-	pubEpoch := b.challenge.PublishEpoch()
-
 	hash, err := b.challenge.Hash()
 	if err != nil {
 		return nil, fmt.Errorf("getting challenge hash failed: %w", err)
 	}
 
 	// the following method waits for a PoET proof, which should take ~1 epoch
-	expireLayer := (pubEpoch + 2).FirstLayer()
-	atxExpired := b.layerClock.AwaitLayer(expireLayer) // this fires when the target epoch is over
+	pubEpoch := b.challenge.PublishEpoch()
+	// Add a 10% margin to the time it takes to generate a PoST proof
+	postDurationWithMargin := time.Duration(float64(b.lastPostGenDuration.Nanoseconds()) * 1.1)
+	postDurationInLayers := b.layerClock.DurationToLayers(postDurationWithMargin)
+	deadlineLayer := b.challenge.PubLayerID.Add(types.GetLayersPerEpoch()).Sub(postDurationInLayers)
+	deadline := b.layerClock.AwaitLayer(deadlineLayer)
```

Comment:
I feel this introduces a possible issue with deadlines. Instead of hard setting a certain layer for when expiration happens now a node can intentionally delay its PoST generation to set a longer deadline?

Reply:
I don't understand what you mean. Could you elaborate?

The longer the PoST takes, the earlier the deadline is set in order to accommodate the time required to run PoST.

Reply:
was is described by someone this way? i don't understand why is this so complex, and why it measures some timings

if we are using a certain instance of poet we know when that poet is expected to terminate. proof should be available almost immediately when poet terminates. so the deadline when we can give up on waiting for that proof is a termination time + any reasonable timeout (5s for example)

Reply:
i think you are overcomplicating things, as well as the description in that thread

there is a poet cycle-gap (delta) which is equal to 5m, this 5m should include and it is important that it needs to be known in advance:
- proof delivery time
- post execution time
- post delivery time

since all this values have to be known in advance - they should be set to the max reasonable value. 

what you are doing by introducing more complexity is dynamically allow to adjust post execution time, in order to increase or decrease proof/post delivery time. it makes literally no difference because whole cycle-gap won't change because of that

Reply:
sorry you are right, lets do what spec suggests

Reply:
OK 👍 It can still be changed later if a better method is found 😉